ext/ffi/lib.rs

```diff
@@ -535,322 +564,792 @@ where
+        ffi_args.push(NativeValue { u64_value: value });
+      }
+      NativeType::I64 => {
+        let value: i64 = if value.is_big_int() {
+          let value = v8::Local::<v8::BigInt>::try_from(value)?;
+          value.i64_value().0
+        } else {
+          value.integer_value(scope).ok_or_else(|| {
+            type_error("Invalid FFI i64 type, expected number")
+          })? as i64
+        };
+        ffi_args.push(NativeValue { i64_value: value });
+      }
+      NativeType::USize => {
+        let value: usize = if value.is_big_int() {
+          let value = v8::Local::<v8::BigInt>::try_from(value)?;
+          value.u64_value().0 as usize
+        } else {
+          value.integer_value(scope).ok_or_else(|| {
+            type_error("Invalid FFI usize type, expected number")
+          })? as usize
+        };
+        ffi_args.push(NativeValue { usize_value: value });
+      }
+      NativeType::ISize => {
+        let value: isize = if value.is_big_int() {
+          let value = v8::Local::<v8::BigInt>::try_from(value)?;
+          value.i64_value().0 as isize
+        } else {
+          value.integer_value(scope).ok_or_else(|| {
+            type_error("Invalid FFI isize type, expected number")
+          })? as isize
+        };
+        ffi_args.push(NativeValue { isize_value: value });
+      }
+      NativeType::F32 => {
+        let value = value
+          .number_value(scope)
+          .ok_or_else(|| type_error("Invalid FFI f32 type, expected number"))?
+          as f32;
+        ffi_args.push(NativeValue { f32_value: value });
+      }
+      NativeType::F64 => {
+        let value = value
+          .number_value(scope)
+          .ok_or_else(|| type_error("Invalid FFI f64 type, expected number"))?
+          as f64;
+        ffi_args.push(NativeValue { f64_value: value });
+      }
+      NativeType::Pointer => {
+        if value.is_null() {
+          let value: *const u8 = ptr::null();
+          ffi_args.push(NativeValue { pointer: value })
+        } else if value.is_big_int() {
+          let value = v8::Local::<v8::BigInt>::try_from(value).unwrap();
+          let value = value.u64_value().0 as *const u8;
+          ffi_args.push(NativeValue { pointer: value });
+        } else if value.is_array_buffer() || value.is_array_buffer_view() {
+          let value: ZeroCopyBuf = serde_v8::from_v8(scope, value)?;
+          let value: &[u8] = &value[..];
+          ffi_args.push(NativeValue {
+            pointer: value.as_ptr(),
+          });
+        } else {
+          return Err(type_error("Invalid FFI pointer type, expected null, BigInt, ArrayBuffer, or ArrayBufferView"));
+        }
+      }
+      NativeType::Function {} => {
+        if value.is_null() {
+          let value: *const u8 = ptr::null();
+          ffi_args.push(NativeValue { pointer: value })
+        } else if value.is_big_int() {
+          // Do we support this? This would be a foreign function pointer given to us by an FFI library.
```

Comment:
Yes, especially now since I moved to using the pointer value even for registered callbacks. This is now any function pointer, including own registered one.